ci/stable_driver.sh

```diff
@@ -73,15 +73,69 @@ $my_dir/stable_mark.sh $stableroot/$datestr/GDASApp
 # run the automated testing
 $my_dir/run_ci.sh -d $stableroot/$datestr/GDASApp -o $stableroot/$datestr/output
 ci_status=$?
+total=0
 if [ $ci_status -eq 0 ]; then
-  # push a new commit to the stable branch
-  cd $stableroot/$datestr/GDASApp
-  git push origin --delete feature/stable-nightly
-  git checkout -b feature/stable-nightly
+  # copy the CMakeLists file for safe keeping
+  cp $stableroot/$datestr/GDASApp/CMakeLists.txt $stableroot/$datestr/GDASApp/CMakeLists.txt.new
+  total=$(($total+$?))
+  if [ $total -ne 0 ]; then
+    echo "Unable to cp CMakeLists" >> $stableroot/$datestr/output
+  fi
+  # checkout feature/stable-nightly
+  git stash
+  total=$(($total+$?))
+  if [ $total -ne 0 ]; then
+    echo "Unable to cp CMakeLists" >> $stableroot/$datestr/output
+  fi
+  git checkout feature/stable-nightly
+  total=$(($total+$?))
+  if [ $total -ne 0 ]; then
+    echo "Unable to cp CMakeLists" >> $stableroot/$datestr/output
+  fi
+  # merge in develop
+  git merge develop
+  total=$(($total+$?))
+  if [ $total -ne 0 ]; then
+    echo "Unable to merge develop" >> $stableroot/$datestr/output
+  fi
+  # force move the copy to the original path of CMakeLists.txt
+  /bin/mv -f $stableroot/$datestr/GDASApp/CMakeLists.txt.new $stableroot/$datestr/GDASApp/CMakeLists.txt
+  total=$(($total+$?))
+  if [ $total -ne 0 ]; then
+    echo "Unable to mv CMakeLists" >> $stableroot/$datestr/output
+  fi
+  # commit this change and push
   git add CMakeLists.txt
+  total=$(($total+$?))
+  if [ $total -ne 0 ]; then
+    echo "Unable to add CMakeLists to commit" >> $stableroot/$datestr/output
+  fi
   git commit -m "Update to new stable build on $datestr"
+  total=$(($total+$?))
+  caution=""
+  if [ $total -ne 0 ]; then
```

Comment:
This exit code is the troublemaker. Should I just remove lines 114-119? Or do we live with the emails @RussTreadon-NOAA @guillaumevernieres

Reply:
Or do we want to echo a date/time to a file to ensure there is always a commit?

Reply:
@CoryMartin-NOAA , I'm doing what you probably already did - searching stackoverflow  to see how others handle this.   There are a few hits (e.g., [how-to-git-commit-nothing-without-an-error](https://stackoverflow.com/questions/8123674/how-to-git-commit-nothing-without-an-error)), but I'm not sure if we want to try any.   

We want to capture errors if something goes wrong when we commit real change(s). 
 We don't want to capture _nothing to commit_ as a error.

Reply:
Yep, my current thought is to do something like:
```
date > commit_generated
git add commit_generated
```
to somewhere near lines 108

Reply:
I'm fine with the darth-vader email @CoryMartin-NOAA .